internal/telemetry/metric_exporter_config.go

```diff
@@ -0,0 +1,109 @@
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package telemetry
+
+import (
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/aggregation"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+)
+
+// Override default otel/prometheus boundaries, as we skip empty buckets and therefore able to use more accurate and higher range boundaries.
+// See https://github.com/elastic/apm/blob/5791498e9569ef9111615ef439e1cbf0b7fd7c18/specs/agents/metrics-otel.md#histogram-aggregation
+var customHistogramBoundaries = []float64{
+	0.00390625, 0.00552427, 0.0078125, 0.0110485, 0.015625, 0.0220971, 0.03125,
+	0.0441942, 0.0625, 0.0883883, 0.125, 0.176777, 0.25, 0.353553, 0.5, 0.707107,
+	1, 1.41421, 2, 2.82843, 4, 5.65685, 8, 11.3137, 16, 22.6274, 32, 45.2548, 64,
+	90.5097, 128, 181.019, 256, 362.039, 512, 724.077, 1024, 1448.15, 2048,
+	2896.31, 4096, 5792.62, 8192, 11585.2, 16384, 23170.5, 32768, 46341.0, 65536,
+	92681.9, 131072,
+}
+
+type Config struct {
+	MetricFilter        []string
+	TemporalitySelector metric.TemporalitySelector
+	AggregationSelector metric.AggregationSelector
+}
+type ConfigOption func(Config) Config
+
+func newConfig(opts ...ConfigOption) Config {
+	config := Config{}
+	for _, opt := range opts {
+		config = opt(config)
+	}
+
+	if config.TemporalitySelector == nil {
+		config.TemporalitySelector = defaultTemporalitySelector
+	}
+	if config.AggregationSelector == nil {
+		config.AggregationSelector = defaultAggregationSelector
+	}
+
+	return config
+}
+
+func defaultTemporalitySelector(ik metric.InstrumentKind) metricdata.Temporality {
+	switch ik {
+	case metric.InstrumentKindCounter, metric.InstrumentKindObservableCounter, metric.InstrumentKindHistogram:
+		return metricdata.DeltaTemporality
+	}
+	return metric.DefaultTemporalitySelector(ik)
+}
+
+func defaultAggregationSelector(ik metric.InstrumentKind) aggregation.Aggregation {
+	switch ik {
+	case metric.InstrumentKindHistogram:
+		return aggregation.ExplicitBucketHistogram{
+			Boundaries: customHistogramBoundaries,
+			NoMinMax:   false,
+		}
+	default:
+		return metric.DefaultAggregationSelector(ik)
+	}
+}
+
+// WithMetricFilter configured the metrics filter. Any metric filtered here
+// will be the only ones to be exported. All other metrics will be ignored.
+//
+// Defaults to not exporting anything.
+func WithMetricFilter(f []string) ConfigOption {
+	return func(cfg Config) Config {
+		cfg.MetricFilter = f
+		return cfg
+	}
+}
```

Comment:
Should we be using Views? e.g. https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric#example-NewView-Drop

Reply:
A view is applied to every instrumentation which matches it. So if we used it, it would filter the metrics for every exporter, not just this one.
We don't want to filter metrics that way for apmotel.

Reply:
@dmathieu explained on slack that we may want to use a single MeterProvider with two readers, in which case that wouldn't work.